Src/WitsmlExplorer.Frontend/components/Modals/AnalyzeGapModal.tsx

```diff
@@ -0,0 +1,116 @@
+
+  const onSubmit = async () => {
+    setIsLoading(true);
+    const logReference: LogObject = props.logObject;
+    const analyzeGapsJob = {
+      logReference: logReference,
+      mnemonics: props.mnemonics,
+      gapSize: gapSize,
+      timeGapSize: convertToMilliseconds(timeGapSize)
+    };
+    const jobId = await JobService.orderJob(JobType.AnalyzeGaps, analyzeGapsJob);
+    const reportModalProps = { jobId };
+    setIsLoading(false);
+    dispatchOperation({ type: OperationType.HideModal });
+    dispatchOperation({ type: OperationType.DisplayModal, payload: <ReportModal {...reportModalProps} /> });
+  };
+
+  const handleGapSizeChanged = (event: React.ChangeEvent<HTMLInputElement>) => {
+    if (event.target.value) {
+      const value = Number(event.target.value);
+      setGapSize(value);
+      setGapSizeIsValid(value >= 0);
+      setGapSizeShowWarning(value >= 0 && value < 1);
+    }
+  };
+
+  const handleTimeGapSizeChanged = (event: React.ChangeEvent<HTMLInputElement>) => {
+    setGapSizeIsValid(new RegExp(timePattern).test(event.target.value));
+    if (event.target.value) {
+      setTimeGapSize(event.target.value);
+      const millisecondsTime = convertToMilliseconds(event.target.value);
+      setGapSizeShowWarning(millisecondsTime >= 0 && millisecondsTime < 60000);
+    }
+  };
+
+  const convertToMilliseconds = (time: string): number => {
+    const [hours, minutes, seconds] = time.split(":").map(Number);
+    return (hours * 3600 + minutes * 60 + seconds) * 1000;
+  };
+
+  return (
+    <>
+      {log && (
+        <ModalDialog
+          heading={`Analyze gaps`}
+          content={
+            <>
+              {log.indexType === WITSML_INDEX_TYPE_DATE_TIME && (
+                <TextField
+                  id={"timeGapSize"}
+                  label="gap size"
+                  value={timeGapSize}
+                  onChange={handleTimeGapSizeChanged}
+                  placeholder="hh:mm:ss"
+                  fullWidth
+                  inputProps={{ maxLength: 8 }}
+                  error={!gapSizeIsValid}
+                  helperText={"Gap size time must be in format hh:mm:ss, where hours: 00..23, minutes: 00..59, seconds: 00..59"}
+                />
+              )}
+              {log.indexType === WITSML_INDEX_TYPE_MD && (
+                <TextField
+                  id={"gapSize"}
+                  label="gap size"
+                  type="number"
+                  fullWidth
+                  value={gapSize}
+                  onChange={handleGapSizeChanged}
+                  error={!gapSizeIsValid}
+                  helperText={"Gap size number must be greater than or equal to 0"}
+                />
+              )}
+              {gapSizeShowWarning && <WarningBar message={"The gap size is too small, gap analysis might take some time."} />}
+            </>
+          }
+          onSubmit={() => onSubmit()}
+          isLoading={isLoading}
+          confirmColor={"danger"}
```

Comment:
Use the default confirmColor (by not setting it)

Reply:
fixed